ci_framework/roles/libvirt_manager/tasks/reserve_ip.yml

```diff
@@ -0,0 +1,51 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# The below tasks reserves an IP from the dhcp range and assigns it to the
+# specified mac address and vm
+
+# The below information are required
+#   vm_name       Name of the virtual machine/domain
+#   mac_address   Associated MAC address
+#   network
+#     name        virtual network name
+#     cidr        network notation
+
+# The nth host is determined by looking to the vm_name
+#   master          20
+#   worker          23
+#   extraworker     26
+#   compute         26      # Has the same index of extraworker
+
+
+- name: Generate the IP address
+  vars:
+    _vm_name: "{{ vm_name | replace('-', '_') }}"
+    _split_vm_name: "{{ vm_replace | split('_') }}"
+    _ip_index: >-
+      {{
+        cifmw_libvirt_manager_vm_net_ip_set[_split_vm_name[1]] +
+        _split_vm_name[2] | int
+      }}
+  ansible.builtin.set_fact:
+    _lm_ip_address: "{{ network.cidr | ansible.utils.nthhost(_ip_index) }}"
```

Comment:
never used that filter - what's the kind of IP it will output then? i.e. what's the compute-0 IP?

Reply:
This filter results the resulting IP address in that index

For example
network 192.168.10.0/24

nthhost(2) -> 192.168.10.2
nthhost(220) -> 192.168.10.220

nthhost(300) # would result in failure.

[Reference](https://docs.ansible.com/ansible/latest/collections/ansible/utils/nthhost_filter.html)

Reply:
compute-0 IP would be `26 + 0` i.e. 192.168.122.26

Reply:
hmmm so it should be 100 - because computes are usually 100, 101, 102, ... :)

Reply:
So we should have 
`controller: 10`
`worker: 13`
`compute: `00`

In the latest commit.